libpod/runtime_img.go

```diff
@@ -48,6 +48,22 @@ func (r *Runtime) RemoveContainersForImageCallback(ctx context.Context) libimage
 	}
 }
 
+// IsExternalContainerCallback returns a callback that be used in `libimage` to
+// figure out whether a given container is an external one.  A container is
+// considered external if it is not present in libpod's database.
+func (r *Runtime) IsExternalContainerCallback(_ context.Context) libimage.IsExternalContainerFunc {
+	return func(idOrName string) (bool, error) {
+		_, err := r.LookupContainer(idOrName)
```

Comment:
This seems prone to races - what if the container was legitimately removed, instead of never existing?

Reply:
I fear that there we'll always face such a race in this specific use case unless pruning is moved to c/storage :(

Reply:
Ah, actually I can work around that. Will update on Monday :)

Reply:
Fresh Monday morning brain tells me that I cannot work around that. We had to move pruning into c/storage for that use case.

In the end, I do not think the race is too bad. If the container has been removed while pruning, it would be considered as an external one and would be attempted to be removed.